csharp-basic-topics/GuidClassInCSharp/GuidClassInCSharp/Methods/GuidToByteString.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace GuidClassInCSharp
+{
+    partial class GuidClassInCSharpMethods
+    {
+        public static string GuidToByteString(Guid inputGuid)
+        {
+            var byteArray = inputGuid.ToByteArray();
```

Comment:
Since you are targetting .NET 7, you can use a stackalloc Span here and call the guid.TryWriteBytes to save on the array allocation.
Span<byte> dest = stackalloc byte[16];
inputGuid.TryWriteBytes(dest);

Reply:
Here, I wanted to demonstrate the .ToByteArray() extension explicitly, and I tried to keep it simple. Should I leave it as it is or add a new section with the extension .TryWriteBytes() ?

Reply:
In my  opinion both are simple and the latter approach (the one I suggested) gives an extra nudge to noalloc/lowalloc code writing, which is an important factor when talking about microservices, mobile devices, embedded systems, etc. Basically, my feeling is even if we are on a desktop, we are still gaining performance and reducing allocation overhead so why not? When people start looking at doing things like .NET MAUI then you have no idea where the eventual code may run. I do understand the trap of early optimization, but in this case it is basically the same code, just with an allocation removed, so we aren't prematurely optimizing and we are not complicating the code. The only caveat with this sample would be if someone was calling it in a recursive loop, but someone who saw this code and wanted to use it in that context would already be aware of that and could make the change to use ToByteArray themselves.

That is my feeling - you should check it with your editor, I am just adding some basic counsel.